### DISCUSSION TITLE
Unable to use background image

### DISCUSSION THREAD

**Discussion by slackermorris** [tailwindlabs/tailwindcss#10387] [Help] [answered]
Hi all,

```javascript
// running versions
"tailwindcss": "^3.0.23",
```

I'm at a complete loss. I am trying to include an image as a background image and have tried a variety of approaches to no avail. I have tried:

- [Customising](https://tailwindcss.com/docs/background-image#customizing-your-theme) the Tailwind theme to refer to my own background image.
```javascript
// tailwind.config.js
backgroundImage: {
      "image-of-banana": "url('../public/image-of-banana.png')",
},
```
- [Using inline arbitrary values](https://tailwindcss.com/docs/background-image#customizing-your-theme) to refer to the same image.
```javascript
// app/routes/bananaPage
<div className="... bg-[url('../../public/image-of-banana.png')] bg-cover bg-center bg-no-repeat ..." />
```

Nothing works. I get a `404 Not Found`, for the image. With the request URL requesting `http://localhost:3000/public/image-of-banana.png`. 

Strangely, requesting the same image via the `<img/>` `src` tag does work. I will also note that I haven't resorted to declaring a custom class at the level of my entry .css file because I am hoping that Tailwind and what it purports to offer in this space will take care of my needs 😅.  

Could someone please help me to find the error in my ways? Thanks for all the help.

**References**
- [Exactly the same issue](https://www.reddit.com/r/tailwindcss/comments/jsgnzh/background_image_problem/). Unfortunately I don't understand the solution.
- [Same issue](https://stackoverflow.com/questions/73812990/tailwind-css-does-not-load-background-images-in-react-app). Solutions don't help me.
- [Ditto as above](https://stackoverflow.com/questions/64742287/background-image-not-showing-up-with-react-tailwindcss).
- [Ditto as above](https://stackoverflow.com/questions/71962510/bg-url-in-tailwindcss-is-not-working-for-me).
- [Ditto as above](https://stackoverflow.com/questions/70731071/background-image-not-showing-up-on-style-css-with-tailwind-css).
- [Ditto as above](https://stackoverflow.com/questions/68652516/tailwind-css-config-file-background-image-issue).
- [Ditto as above](https://stackoverflow.com/questions/63658218/tailwind-css-backgroundimage-doesnt-work-for-me).
- [Ditto as above](https://github.com/tailwindlabs/tailwindcss/discussions/6617). And why should we have to use a style tag anyway?

**Comment by wongjn** [comment]
Consider matching the path inside the `url()` with the working `src` value of the `<img>` element.

**Comment by adamwathan** [comment, marked as answer]
Make sure the URL matches the actual URL where the image is being served from, not the path to the image on your local machine. Usually the public directory is just served from root on your server, so my guess is the right path is just `url(‘/image-of-banana.ping’)` or maybe `url(‘/public/img-of-banana.png’)`.


If you are going to ever use relative URLs (I would recommend against it), they should be relative to where the compiled CSS file lives, not relative to your template file.

**Reply by iqbalpa** [reply]
thank you very much

**Reply by mrdulin** [reply]
This should be added to the docs.
